src/mixins/java/org/spongepowered/common/mixin/tracker/server/management/PlayerInteractionManagerMixin_Tracker.java

```diff
@@ -95,5 +135,71 @@ public boolean afterOpenContainer(BlockState blockState, World worldIn, PlayerEn
         return result;
     }
 
+    @Inject(method = "processRightClick", cancellable = true,
+            at = @At(value = "INVOKE",
+                    target = "Lnet/minecraft/item/ItemStack;getCount()I"))
+    public void onRightClick(PlayerEntity player, World worldIn, ItemStack stack, Hand hand, CallbackInfoReturnable<ActionResultType> cir) {
+        InteractItemEvent.Secondary event = SpongeCommonEventFactory.callInteractItemEventSecondary(player, stack, hand, null, null);
+        if (event.isCancelled()) {
+            cir.setReturnValue(ActionResultType.FAIL);
+        }
+    }
+
+    @Inject(method = "func_225416_a", cancellable = true, at = @At(value = "HEAD"))
+    public void onLeftClickBlock(BlockPos p_225416_1_, CPlayerDiggingPacket.Action p_225416_2_, Direction p_225416_3_, int p_225416_4_, CallbackInfo ci) {
+        BlockSnapshot snapshot = ((ServerWorld) (this.world)).createSnapshot(VecHelper.toVector3i(p_225416_1_));
+        InteractBlockEvent.Primary event = SpongeCommonEventFactory.callInteractBlockEventPrimary(this.player, this.player.getHeldItem(Hand.MAIN_HAND), snapshot, Hand.MAIN_HAND, p_225416_3_, null);
+        if (event.isCancelled()) {
+            ci.cancel();
+        }
+    }
+
+    /**
+     * @author Morph
+     * @reason Fire interact block event.
+     */
+    @Overwrite
+    public ActionResultType func_219441_a(PlayerEntity playerIn, World worldIn, ItemStack stackIn, Hand handIn, BlockRayTraceResult blockRaytraceResultIn) {
+        BlockPos blockpos = blockRaytraceResultIn.getPos();
+        BlockState blockstate = worldIn.getBlockState(blockpos);
+        // Sponge start
+        BlockSnapshot snapshot = ((ServerWorld) (worldIn)).createSnapshot(VecHelper.toVector3i(blockpos));
+        Vector3d hitVec = Vector3d.from(blockRaytraceResultIn.getHitVec().getX(), blockRaytraceResultIn.getHitVec().getY(), blockRaytraceResultIn.getHitVec().getZ());
+        InteractBlockEvent.Secondary event = SpongeCommonEventFactory.callInteractBlockEventSecondary(playerIn, stackIn, hitVec, snapshot, null, handIn);
```

Comment:
With this being an overwrite, we MUST call a `PlatformHook` here so that Forge's changes can be honored.

That or we state right now that Forge's interact event will mesh 1:1 with ours.

Need to decide @dualspiral @ImMorpheus

Reply:
@dualspiral

Reply:
> That or we state right now that Forge's interact event will mesh 1:1 with ours.

I don't want to guarantee that. When we can afford ourselves the flexibility to change the event, we should. We don't know 100% how the phase tracker will handle anything here yet (especially as this is placed in the tracker mixin package, should it be?).

Reply:
Another reason for the `PlatformHook` would allow us to preserve the overwrite here in common and not cause Mixin-ception by having Forge overwrite this.

This is all contingent on if our event is mixed onto Forge's.

Reply:
This is actually true for all of these and our overall strategy with Forge.

If the injections remain we'll fire our event and then theirs.

Reply:
> With this being an overwrite, we MUST call a PlatformHook here so that Forge's changes can be honored.

Also, Forge adds a doesSneakBypass method later. A PlatformHook for that is needed: https://github.com/MinecraftForge/MinecraftForge/blob/5224533d45ad820121deb09f2419b8680f7b5f3b/patches/minecraft/net/minecraft/server/management/PlayerInteractionManager.java.patch#L146

> That or we state right now that Forge's interact event will mesh 1:1 with ours.

:+1: for deciding on a strategy on sponge-forge event interaction

> (especially as this is placed in the tracker mixin package, should it be?)

I've added them there because that's where `InteractContainerEvent.Open` is placed.

Reply:
> If the injections remain we'll fire our event and then theirs.

@Zidane Do we fire their events if our is cancelled ? 
Also, do we fire the event again after the forge one ?